h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2339,6 +2357,49 @@ else if (gaussian.equals(_parms._family) && Link.identity.equals(_parms._link))
       }
     }
 
+    /***
+     * Estimate dispersion factor using maximum likelihood.  I followed section IV of the doc in 
+     * https://h2oai.atlassian.net/browse/PUBDEV-8683 . 
+     */
+    public double estimateMLSE(GLMTask.ComputeMLSETsk mlCT, double alpha, double[] beta) {
+      double constantValue = mlCT._wsum + mlCT._sumlnyiOui - mlCT._sumyiOverui;
+      DataInfo dinfo = _state.activeData();
+      Frame adaptedF = dinfo._adaptedFrame;
+      int count = 0;
+      long currTime = System.currentTimeMillis();
+      long modelBuiltTime = currTime - _model._output._start_time;
+      
+      // stopping condition for while loop are:
+      // 1. magnitude of iterative change to se < EPS
+      // 2. for every 100th iteration, we check for additional stopping condition:
+      //    a.  User requests stop via stop_requested;
+      //    b.  time taken to find the dispersino parameter has exceeded time taken to build the GLM model.
+      while (true) {
+        GLMTask.ComputeDiTriGammaTsk ditrigammatsk = new GLMTask.ComputeDiTriGammaTsk(null, dinfo, _job._key, beta,
+                _parms, alpha).doAll(adaptedF);
+        double numerator = mlCT._wsum*Math.log(alpha)-ditrigammatsk._sumDigamma+constantValue; // equation 2 of doc
+        double denominator = mlCT._wsum/alpha - ditrigammatsk._sumTrigamma;  // equation 3 of doc
+        double change = numerator/denominator;
+        if (denominator == 0 || Double.isNaN(change))
+          return alpha;
+        if (Math.abs(change) < EPS) // stop if magnitude of iterative updates to se < EPS
+          return alpha-change;
+        else {
+          count++;
+          double se = alpha - change;
+          if (se < 0) // heuristc to prevent seInit <= 0
+            alpha *= 0.5;
+          else
+            alpha = se;
+        }
+
+        if ((count % 100 == 0) && // check for additional stopping conditions for every 100th iterative steps
+            (stop_requested() ||  // user requested stop via stop_requested()
+            (System.currentTimeMillis()-currTime) > modelBuiltTime)) // time taken to find dispersino exceeds GLM model building time
```

Comment:
I will think should produce a warning when we stop based on time (the warning should say the calculation is non-deterministic and results are approximated)